docs/sql/recursion.mdx

```diff
@@ -0,0 +1,251 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Mutually-recursive queries
+
+:::warning
+
+Recursive queries are a new feature in Feldera SQL and are still evolving. Syntax and
+functionality may change in future versions.
+
+:::
+
+Recursive computations are notoriously limited in standard SQL due to cumbersome syntax and limited expressivity.
+Feldera SQL introduces an intuitive and powerful syntax for recursive queries that:
+
+- Removes the need for Common Table Expressions (CTEs) in recursive queries.
+- Supports mutually recursive view definitions involving multiple views.
+- Allows rich, non-monotonic queries in recursive view definitions, with a goal to support arbitrary queries in the future.
+
+These enhancements make Feldera SQL Turing-complete, meaning recursive queries that never terminate can be written—so
+care must be taken when designing queries.
+
+## Key Concepts of Recursive Views
+
+### Defining Recursive Views with Forward Declarations
+
+In Feldera SQL, recursive queries require [forward declarations](https://en.wikipedia.org/wiki/Forward_declaration)
+when a view references itself or another view before their definition.
+A forward declaration specifies the view's name, column names, and column types.
+
+```sql
```

Comment:
The text explained the purpose of this view.
Not explaining it makes it harder to read.

Reply:
Are you referring to the text that talked about forward declaration in C?

Reply:
Edges in a graph